### PR TITLE
JDK-8319413: Start of release updates for JDK 23

### PR DESCRIPTION
**Pull request by jddarcy**
Time to start making preparations for JDK 23.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change requires CSR request [JDK-8319428](https://bugs.openjdk.org/browse/JDK-8319428) to be approved
- [x] Change requires CSR request [JDK-8319426](https://bugs.openjdk.org/browse/JDK-8319426) to be approved
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8319427](https://bugs.openjdk.org/browse/JDK-8319427) to be approved

### Issues
 * [JDK-8319413](https://bugs.openjdk.org/browse/JDK-8319413): Start of release updates for JDK 23 (**Bug** - P4)
 * [JDK-8319414](https://bugs.openjdk.org/browse/JDK-8319414): Add SourceVersion.RELEASE_23 (**Sub-task** - P4)
 * [JDK-8319416](https://bugs.openjdk.org/browse/JDK-8319416): Add source 23 and target 23 to javac (**Sub-task** - P4)
 * [JDK-8319428](https://bugs.openjdk.org/browse/JDK-8319428): Start of release updates for JDK 23 (**CSR**)
 * [JDK-8319427](https://bugs.openjdk.org/browse/JDK-8319427): Add SourceVersion.RELEASE_23 (**CSR**)
 * [JDK-8319426](https://bugs.openjdk.org/browse/JDK-8319426): Add source 23 and target 23 to javac (**CSR**)

### Reviewers
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**) ⚠️ Review applies to [2ee6cb96](https://git.openjdk.org/jdk/pull/16505/files/2ee6cb968ee7e9b2d45fdfae4b34bea0bcd20b08)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [a0bc033d](https://git.openjdk.org/jdk/pull/16505/files/a0bc033d3618531bbdd8311ee08cb6cab23a3f07)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**) ⚠️ Review applies to [9daf1014](https://git.openjdk.org/jdk/pull/16505/files/9daf1014287cd6647007e750c31654957aea3d3f)
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**) ⚠️ Review applies to [9daf1014](https://git.openjdk.org/jdk/pull/16505/files/9daf1014287cd6647007e750c31654957aea3d3f)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16505/head:pull/16505` \
`$ git checkout pull/16505`

Update a local copy of the PR: \
`$ git checkout pull/16505` \
`$ git pull https://git.openjdk.org/jdk.git pull/16505/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16505`

View PR using the GUI difftool: \
`$ git pr show -t 16505`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16505.diff">https://git.openjdk.org/jdk/pull/16505.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16505#issuecomment-1828418241)

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/classfile/Classfile.java
Comment (liach) [inline src/java.base/share/classes/jdk/internal/classfile/Classfile.java:1]: `latestMajorVersion` below should be updated to return `JAVA_23_VERSION` as well.

**Review by liach — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/jdk/internal/classfile/Classfile.java
Comment (jddarcy) [reply src/java.base/share/classes/jdk/internal/classfile/Classfile.java:1]: Good catch; fixed.

**Comment by jddarcy** [comment]
Nothing exception in this batch of start-of-release updates. Clean local testing results on tier 1.

The symbol files correspond to JDK 22 build 25 and will be updated with subsequent builds as usual.

**Review by irisclark — APPROVED** [review]

**Review by dholmes-ora — COMMENTED** [review]
Hotspot change is trivially fine, but too insignificant to hit the Approve button.

Thanks

**Review by erikj79 — APPROVED** [review]
Build changes look good.

**Review by irisclark — APPROVED** [review]

**Review by AlanBateman — APPROVED** [review]
Looks good. I assume you'll bump the copyright header on the files that need it before integration, e.g. JavacTestingAbstractProcessor.

**Review by vicente-romero-oracle — APPROVED** [review]
looks good to me

File: test/langtools/tools/javac/versions/Versions.java [hunk 1]
```diff
@@ -93,6 +94,7 @@ static enum SourceTarget {
         TWENTY(false,    "64.0", "20"),
         TWENTY_ONE(false,"65.0", "21"),
         TWENTY_TWO(false,"66.0", "22"),
+        TWENTY_THREE(false,"67.0", "23"),
```
Comment (turbanoff) [inline test/langtools/tools/javac/versions/Versions.java:97]: As alignment is broken anyway, let's add space
```suggestion
        TWENTY_THREE(false, "67.0", "23"),
```

Comment (irisclark) [reply test/langtools/tools/javac/versions/Versions.java:95]: If you're going to add the space, I'd consider establishing new alignment at TWENTY to accommodate the eventual space needed for "THREE", "SEVEN", and "EIGHT".  You wouldn't need to reset alignment for quite some time after that.

**Review by irisclark — APPROVED** [review]

**Pull request closed without merging**